### ISSUE TITLE
Improving the Typing Class

### ISSUE THREAD

**Issue by lorentzenman** [SpiderLabs/sheepl#12]
TLDR; Push the code for the new typing module I've been working on and then update all the modules to use this new object.

## Background 
Atm, all keystrokes and therefore the captures are perfect. I've been working on a new typing object that breaks that. Here is a quick list of the features:
* Thresholds for typing ability (really bad to epic)
* Ability to accent certain elements, e.g. bold words in Word etc. These will need to be task specific. Easier to just attach the capabilities to the object and then to work out whether they need to be used in the respective task.
* Copy and pasting to and from things. Selecting all text etc.

Overall the goal is to take the current output you get from a key capture like this:
"I am the best typist in the world"....
and turn it into this....
"I an<backspace>m rhn<backspace><backspace><backspace>the b3s<backspace><backspace>est typist in the world"
